### ISSUE TITLE
Error when trying to convert from GrayAlpha to Rgba

### ISSUE THREAD

**Issue by Kaktusbot** [awxkee/moxcms#142]
https://github.com/awxkee/moxcms/blob/b7b4c8ba0d9f0fe083a727b4f2921f963149e83d/src/conversions/gray2rgb.rs#L364-L379

I don't understand anything in the code above, but my small program below was made to convert every single image to Rgba and it was getting errors for some rare gray images in my library.
```rust
...
let transform = color_profile.create_transform_8bit(
            match (meta.num_color_channels, meta.has_alpha_channel) {
                (3, false) => moxcms::Layout::Rgb,
                (3, true) => moxcms::Layout::Rgba,
                (1, false) => moxcms::Layout::Gray,
                (1, true) => moxcms::Layout::GrayAlpha,
                _ => return Err("Unsupported!".into()),
            },
            &dest_profile,
            moxcms::Layout::Rgba,
            moxcms::TransformOptions::default(),
        )?;
...
```

The error was "LaneSizeMismatch". After I patched line 379 to be  `{ Layout::GrayAlpha as u8 },` it stopped throwing errors and the image transformed successfully. Lines 367 and 368 both seem to be incorrect too. Line 391 as well.
I have no idea how Alpha is handled, maybe there's some other place in the code that handles alpha and I'm wrong about my assumptions. But then I don't think it should throw errors on a valid transform, maybe? I don't know. Maybe I'm just doing something wrong. After all, I can see this code is 10 months old and nobody complained about it.

**Comment by awxkee** [comment]
Thanks for reporting!

This is a typo, should be fixed after #143
